jax/numpy/lax_numpy.py

```diff
@@ -1664,6 +1664,84 @@ def identity(n, dtype=None):
   lax._check_user_dtype_supported(dtype, "identity")
   return eye(n, dtype=dtype)
 
+@_wraps(onp.unique)
+def unique(ar, return_index=False, return_inverse=False,return_counts=False,
+           axis=None):
+
+	def _unpack_tuple(x):
+		if len(x) == 1:
+			return x[0]
+		else:
+			return x
+
+	def asanyarray(a, dtype=None, order=None):
+	    return onp.array(a, dtype, copy=False, order=order, subok=True)
+
+	def ascontiguousarray(a, dtype=None):
+	    return onp.array(a, dtype, copy=False, order='C', ndmin=1)
+
+	def _unique1d(ar, return_index=False, return_inverse=False,
+				return_counts=False):
+
+		ar = asanyarray(ar).flatten()
+
+		optional_indices = return_index or return_inverse
+
+		if optional_indices:
+			perm = ar.argsort(kind='mergesort' if return_index else 'quicksort')
+			aux = ar[perm]
+		else:
+			ar.sort()
+			aux = ar
+		mask = onp.empty(aux.shape, dtype=onp.bool_)
+		mask[:1] = True
+		mask[1:] = aux[1:] != aux[:-1]
+
+		ret = (aux[mask],)
+		if return_index:
+			ret += (perm[mask],)
+		if return_inverse:
+			imask = onp.cumsum(mask) - 1
+			inv_idx = onp.empty(mask.shape, dtype=onp.intp)
+			inv_idx[perm] = imask
+			ret += (inv_idx,)
+		if return_counts:
+			idx = onp.concatenate(onp.nonzero(mask) + ([mask.size],))
+			ret += (onp.diff(idx),)
+		return (ret)
+
+	ar = asanyarray(ar)
```

Comment:
Unfortunately it's not a good idea to cast to a classic NumPy array (here, you call `onp.array` on the input, which yields a NumPy).

e.g.:
```
In [26]: jax.numpy.ones((3, 4)) + 2                                                                       
Out[26]: 
DeviceArray([[3., 3., 3., 3.],
             [3., 3., 3., 3.],
             [3., 3., 3., 3.]], dtype=float32)

In [27]: asanyarray(jax.numpy.ones((3, 4)) + 2)                                                           
Out[27]: 
array([[3., 3., 3., 3.],
       [3., 3., 3., 3.],
       [3., 3., 3., 3.]], dtype=float32)
```
Note the output is a classic NumPy array.

The problem with doing this is that JAX's tracer doesn't know understand how to trace through classic NumPy ops, only the `lax` primitives and operations built on top of them. So, for example, you wouldn't be able to batch this operation with `vmap` or compile it with `jit`. So you need to express this operation in terms of the operations in `jax.lax` (or other `jax.numpy` operations).

I also note that if we *did* want to implement this operation by casting to a classic NumPy array, then there's a much simpler version...
```
unique = onp.unique
```

Reply:
Hey th

Reply:
Hey there!  In functions like _get_min_identity(dtype) the return type is onp.array therefore i used in the above function. Would it be viable to just typecast the return type in a normal list or use lax.full() with appropriate parameters .